source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -657,6 +662,12 @@ The event API here is assumed to be like the standard `Python Event
         heartbeatFrequencyMS = heartbeatFrequencyMS
         minHeartbeatFrequencyMS = 500
 
+        # Heartbeat command
+        if versionedApi or handshakeResponse.helloOk:
```

Comment:
There won't be a `handshakeResponse` available in the monitor's constructor since the first legacy hello hasn't been called yet. And if the server is marked unknown, I think we'll want to send a legacy hello command again with `helloOk` (in case it was downgraded).

Would it make sense to store a bool on the ServerDescription to indicate whether the server can accept a `hello` command? Something like `ServerDescription.helloOk`?

A response to a legacy hello would set it to true if the response includes `helloOk`. And any response to `hello` would set it to true regardless.

Then, the logic here would follow suit with how `topologyVersion` is handled.

cc @ShaneHarvey in case you'd like to weigh in.

Reply:
Valid point. I've reworked the logic a bit to add `ServerDescription.helloOk` as this is a sensible place to hang this per-server information. The one drawback is that it adds a field to `ServerDescription`, which is a breaking - though additive - change.